core/model/modx/modresponse.class.php

```diff
@@ -76,6 +76,7 @@ public function outputContent(array $options = array()) {
 
             /*FIXME: only do this for HTML content ?*/
             if (strpos($this->contentType->get('mime_type'), 'text/html') !== false) {
+                $this->modx->invokeEvent('OnBeforeRegScripts');
```

Comment:
I think it makes more sense that the event is called `OnBeforeRegisterClientStartupScripts`.

Reply:
I would prefer `OnBeforeRegisterScripts` (or maybe `OnBeforeRegisterClientScripts`)

Reply:
@JoshuaLuckers It fires before adding all the registered scripts not only startup ones.